packages/next/src/server/app-render/app-render.tsx

```diff
@@ -737,9 +737,17 @@ async function renderToHTMLOrFlightImpl(
         hasPostponed,
       })
 
+      function onHeaders(headers: Headers): void {
+        // Copy headers created by React into the response object.
+        headers.forEach((value: string, key: string) => {
+          res.setHeader(key, value)
```

Comment:
Should you not instead use `appendHeader` to ensure that you're not replacing existing headers that are written?

Reply:
Yea that's probably more in spirit with the API.